### PR TITLE
IBX-5644 - Email triggers

### PR DESCRIPTION
**Pull request by dabrt**
| Question      | Answer |
| ------------- | --- |
| JIRA Ticket   | IBX-5644 |
| Versions      | 4.6 |
| Edition       | |

Add a description of email triggers feature.

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/personalization/triggers.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+With e-mail triggers, you can increase the engagement of your visitors and customers by delivering recommendations straight to their mailboxes.
+As a store manager, you can expect that more engaged customers translate to bigger income, improved fulfillment of customer needs, while saving time and effort.
+
+## E-mail trigger types
+
+There are several e-mail triggers available in [[= product_name =]], including the following two:
+
+- Abandoned basket trigger: Personalization engine monitors the user's cart and pushes a message when cart status remains unchanged for a set time. The message contains items that have been abandoned in the cart.
+
+- Reactivation aka. "We miss you" trigger: Personalization engine monitors the user's overall activity and pushes a message when they have not returned to the site for a set time. Recommendations come from the [also purchased and also clicked](recommendation_models.md#also-clicked--purchased) models, while the [top purchased](recommendation_models.md#popularity-models) model is used as fallback.
```
Comment (dkorostynski) [inline docs/personalization/triggers.md:16]: The entire sentence:
Recommendations come from the [also purchased and also clicked](recommendation_models.md#also-clicked--purchased) models, while the [top purchased](recommendation_models.md#popularity-models) model is used as fallback.

I would like to replace with:
Recommendations are adjusted to the user previous interests, based on the user activity. - something like that.

File: docs/personalization/triggers.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Trigger calculation frequency
+
+Personalization engine checks user context data against trigger conditions every night.
+E-mail triggers are automatically initiated when a specific user's action, inaction or pattern of actions meet certain conditions that are defined in the service.
+
+## Configuring e-mail triggers
+
+E-mail trigger calculations are done on a server that is run and maintained by Ibexa. 
+Also, [[= product_name =]] does not provide a mechanism to send e-mails to the recipient, but delivers a response with recommendations to an Ibexa Connect [webhook](https://doc.ibexa.co/projects/connect/en/latest/tools/webhooks/). Each response includes a list of recommendations, and a user email address that can be used to route a message.
```
Comment (dkorostynski) [inline docs/personalization/triggers.md]: but delivers a response with recommendations to an Ibexa Connect [webhook]
replace with:
but delivers a response with recommendations to teh webhook - for example an Ibexa Connect

File: docs/personalization/triggers.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+- Abandoned basket trigger: Personalization engine monitors the user's cart and pushes a message when cart status remains unchanged for a set time. The message contains items that have been abandoned in the cart.
+
+- Reactivation aka. "We miss you" trigger: Personalization engine monitors the user's overall activity and pushes a message when they have not returned to the site for a set time. Recommendations come from the [also purchased and also clicked](recommendation_models.md#also-clicked--purchased) models, while the [top purchased](recommendation_models.md#popularity-models) model is used as fallback.
+
+## Trigger calculation frequency
+
+Personalization engine checks user context data against trigger conditions every night.
+Email triggers are automatically initiated when a specific user's action, inaction or pattern of actions meet certain conditions that are defined in the service.
+
+Depending on the trigger type and specific settings made for your organization, each email trigger might be initiated several times at set intervals.
```
Comment (kaborkowski) [inline docs/personalization/triggers.md:23]: Interval is not handled yet. I think we have more important features right now (attributes/messages limit), thus I am not sure if will be able to finish it before release. What is more I think we will allow minimum interval 1h - but as I said we have not defined anything in here yet. @dkorostynski what do you think?

**Review by dkorostynski — APPROVED** [review]

**Comment by dabrt** [comment]
Developers accepted PR over slack

File: docs/personalization/triggers.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+## Trigger calculation frequency
+
+Personalization engine checks user context data against trigger conditions every night.
+Email triggers are automatically initiated when a specific user's action, inaction or pattern of actions meet certain conditions that are defined in the service.
+
+## Configuring email triggers
+
+Email trigger calculations are done on a server that is run and maintained by Ibexa. 
+Also, [[= product_name =]] does not provide a mechanism to send emails to the recipient, but delivers a response with recommendations to an endpoint provided by your organization, for example, an Ibexa Connect [webhook](https://doc.ibexa.co/projects/connect/en/latest/tools/webhooks/).
```
Comment (juskora) [inline docs/personalization/triggers.md:28]: Not clear description, some contradictory statements make it hard to understand. 
doesn't, not does not.

File: docs/personalization/triggers.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+Personalization engine checks user context data against trigger conditions every night.
+Email triggers are automatically initiated when a specific user's action, inaction or pattern of actions meet certain conditions that are defined in the service.
+
+## Configuring email triggers
+
+Email trigger calculations are done on a server that is run and maintained by Ibexa. 
+Also, [[= product_name =]] does not provide a mechanism to send emails to the recipient, but delivers a response with recommendations to an endpoint provided by your organization, for example, an Ibexa Connect [webhook](https://doc.ibexa.co/projects/connect/en/latest/tools/webhooks/). 
+Each response includes a list of recommendations, and a user email address that can be used to route a message to the recipient.
+
```
Comment (juskora) [inline docs/personalization/triggers.md:30]: Is the recipient a user with their email mentioned in the response string?

File: docs/personalization/triggers.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+To enable email triggers for your organization, contact your administrator or development team about [preparing a webhook address and processing the response delivered to the webhook]([[= developer_doc =]]/personalization/integrate_recommendation_service/#send-emails-with-recommendations).
```
Comment (juskora) [inline docs/personalization/triggers.md:31]: Maybe put first what should be prepared before contacting dev team.

File: docs/personalization/triggers.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+You can define one or more email triggers of certain type, to support different use cases.
+For each email trigger type, you need to decide on several crucial parameters, for example:
+
+- one or more [types of content](content_types.md)
```
Comment (juskora) [inline docs/personalization/triggers.md:36]: Content Types

File: docs/personalization/triggers.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+- [attributes](recommendation_models.md/#nominal-attributes) to be included in the response
```
Comment (juskora) [inline docs/personalization/triggers.md:37]: what attributes include in the response

File: docs/personalization/triggers.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+- time that must pass before email start being sent
+- number of repetitions
```
Comment (juskora) [inline docs/personalization/triggers.md:39]: repetitions of what?

**Pull request merged**